frontend/src/tests/user-permission-tests.js

```diff
@@ -30,26 +62,54 @@ export function userPermissionsTests(api) {
     });
 
     it("A non-admin cannot access the admin route", async () => {
-        let resp = await apiPOST("/admin/users", newUserData);
-        // Save any extra attributes we got, such as an `id`
-        Object.assign(newUserData, resp.payload);
-
-        // Set the active user to the newly-created user
-        resp = await apiPOST("/debug/active_user", { id: newUserData.id });
+        const prevUserId = await setupUserPerm(newInstructorData);
```

Comment:
Based on this pattern, it might be nicer to have a class that is initialized with a user e.g. named `ActiveUserSwitcher`, and then you can call `await ActiveUserSwitcher.makeActive()` and `await ActiveUserSwitcher.revert()` or something like that.

Reply:
If you create a class for this, it should also have its own test (maybe an existing `active_user` can do double duty?)